web/src/presence.ts

```diff
@@ -189,17 +186,13 @@ export function update_info_from_event(
     const raw = raw_info.get(user_id) ?? {
         server_timestamp: 0,
     };
+    if (server_timestamp !== undefined) {
+        raw.server_timestamp = server_timestamp;
```

Comment:
This probably deserves a comment explaining why we do this. Something like:

> We treat presence events as having arrived at the time of the **previous** poll response from the server, since the event doesn't provide a stronger guarantee for the server's time.